docs/spec/consensus/creating-proposal.md

```diff
@@ -0,0 +1,42 @@
+# Creating a proposal
+
+A block consists of a header, transactions, votes (the commit),
+and a list of evidence of malfeasance (ie. signing conflicting votes).
+
+We include no more than 1/10th of the maximum block size
+(`ConsensusParams.BlockSize.MaxBytes`) of evidence with each block.
+
+## Reaping transactions from the mempool
+
+When we reap transactions from the mempool, we calculate maximum data
+size by subtracting maximum header size (`MaxHeaderBytes`), the maximum
+amino overhead for a block (`MaxAminoOverheadForBlock`), the size of
+the last commit (if present) and evidence (if present). While reaping
+we account for amino overhead for each transaction.
+
+```go
+func MaxDataBytes(maxBytes int64, valsCount, evidenceCount int) int64 {
+	return maxBytes -
```

Comment:
What should we do if maxBytes is less than the sum of `MaxAminoOverheadForBlock+ MaxHeaderBytes+int64(valsCount)*MaxVoteBytes+int64(evidenceCount)*MaxEvidenceBytes`? 
MaxDataBytes will return a negative number. This in turn will lead to all txs being rejected by TxFilter (`int64(len(tx)) <= maxDataBytes`).

Reply:
BlockSize.MaxBytes should account for header size + last commit size + possible evidence.

Right now, the minimum for BlockSize.MaxBytes is `1`.

Reply:
I think it's ok to panic.